leaderelection/config.go

```diff
@@ -129,9 +130,22 @@ type ComponentConfig struct {
 // StatefulSetConfig represents the required information for a StatefulSet service.
 type StatefulSetConfig struct {
 	StatefulSetName string
-	ServiceName     string
-	Port            string
-	Protocol        string
+	ServiceName     string `envconfig:"STATEFUL_SERVICE_NAME" required:"true"`
+	Port            string `envconfig:"STATEFUL_SERVICE_PORT" default:"80"`
+	Protocol        string `envconfig:"STATEFUL_SERVICE_PROTOCOL" default:"http"`
+}
+
+func NewStatefulSetConfig() (*StatefulSetConfig, error) {
+	var ssc StatefulSetConfig
+	ssn, _, err := ParseControllerOrdinal()
```

Comment:
This seems strange we parse the value manually and delegate the rest to the envconfig.
It seems envconfig has methods for fancier parsing, which we might use, but this can be done later.

Reply:
I wonder whether we still need `StatefulSetName` in `StatefulSetConfig`. It's only used in `statefulSetBuilder. BuildElector`, where `ControllerOrdinal` is called. However `ControllerOrdinal` calls `ParseControllerOrdinal` which returns stateful set name as well. I think we can just use `ParseControllerOrdinal`.